corehq/apps/export/models/new.py

```diff
@@ -229,43 +247,101 @@ class Meta:
         app_label = 'export'
 
     @property
+    def is_safe(self):
+        """For compatability with old exports"""
+        return self.is_deidentified
+
+    @property
     def defaults(self):
         return FormExportInstanceDefaults if self.type == FORM_EXPORT else CaseExportInstanceDefaults
 
-    @staticmethod
-    def generate_instance_from_schema(schema, domain, app_id=None):
-        """Given an ExportDataSchema, this will generate an ExportInstance"""
-        instance = ExportInstance(
-            type=schema.type
+    def get_table(self, path):
+        for table in self.tables:
+            if table.path == path:
+                return table
+        return None
+
+    @classmethod
+    def _new_from_schema(cls, schema):
+        raise NotImplementedError()
+
+    @classmethod
+    def update_export_from_schema(cls, schema, saved_export):
+        return cls.generate_instance_from_schema(
+            schema,
+            saved_export.domain,
+            app_id=saved_export.app_id,
+            saved_export=saved_export,
         )
-        instance.name = instance.defaults.get_default_instance_name(schema)
+
+    @classmethod
+    def generate_instance_from_schema(cls, schema, domain, app_id=None, saved_export=None):
+        """Given an ExportDataSchema, this will generate an ExportInstance"""
+        if saved_export:
+            instance = saved_export
+        else:
+            instance = cls._new_from_schema(schema)
+
+        instance.name = instance.name or instance.defaults.get_default_instance_name(schema)
 
         latest_build_ids_and_versions = get_latest_built_app_ids_and_versions(domain, app_id)
         for group_schema in schema.group_schemas:
-            table = TableConfiguration(
+            table = instance.get_table(group_schema.path) or TableConfiguration(
                 path=group_schema.path,
                 name=instance.defaults.get_default_table_name(group_schema.path),
                 display_name=instance.defaults.get_default_table_name(group_schema.path),
                 selected=instance.defaults.default_is_table_selected(group_schema.path),
             )
-            table.columns = map(
-                lambda item: ExportColumn.create_default_from_export_item(
+            columns = []
+            for item in group_schema.items:
```

Comment:
I don't think it's that big a deal:
* single -> multi: any 'extra' options will get put in the 'other' column
* multi-> single: all values in a single column

I think the only issue is if the user started with a single select and then the app changed to a multi-select then their custom export will stay as a single select. If they don't want they then they'd have to remove the column, save the export and then add the column back (assuming that would work).